### PR TITLE
added a web version with no node requires

### PR DESCRIPTION
**Pull request by aqrojo**
Now you can import the web version [if you have errors derived by the fs module](https://github.com/renanbastos93/image-to-base64/issues/17)
````
import imageToBase64 from 'image-to-base64/web'
````

### CODE REVIEW

File: browser.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    function imageToBase64Browser (urlOrImage, param) {
+        if (!("fetch" in window && "Promise" in window)) {
+            return Promise.reject("[*] It's image2base64 not compatible with your browser.");
+        }
+        return fetch(urlOrImage, param || {}).then(function(response){
+            return response.arrayBuffer()
+        }).then(base64ToBrowser);
+    }
+
+    if (typeof module !== "undefined") {
```
Comment (renanbastos93) [inline browser.js:17]: I have a question, even using react we need to validate `module.exports`?

Comment (aqrojo) [reply browser.js:17]: I suppose that the validation isn't required it but I'm not sure if it another libraries may need it, so if you think it's safe we can remove it

**Comment by aqrojo** [comment]
> Good job, thanks for contributing here.

Thanks :) It was so easy, the main work was do it for you

> Well, I believe we need to improve together with this pull request our readme, right?

Of course yes, if you think that the modification is valid, we should update the documentation

> I think too, that this way we can remove little code from `image-to-base64.js` file.

Yes, my first approach was to separate web code from the node one, but I retracted that idea because those changes might seem a little intrusives and may be good to maintain backward compatibility, anyway I believe that's a wise idea

Now that I know what you think about it, I can send you some updates next Monday

**Comment by renanbastos93** [comment]
> > Good job, thanks for contributing here.
> 
> Thanks :) It was so easy, the main work was do it for you
> 
> > Well, I believe we need to improve together with this pull request our readme, right?
> 
> Of course yes, if you think that the modification is valid, we should update the documentation
> 
> > I think too, that this way we can remove little code from `image-to-base64.js` file.
> 
> Yes, my first approach was to separate web code from the node one, but I retracted that idea because those changes might seem a little intrusives and may be good to maintain backward compatibility, anyway I believe that's a wise idea
> 
> Now that I know what you think about it, I can send you some updates next Monday

How nice, I liked. Let's go to improve it.

Well, I am too made some commits here please to make a rebase.

File: browser.js (hunk 1 above)
Comment (renanbastos93) [reply browser.js:17]: Let's awaiting, for now, later we go remove, right?

**Comment by aqrojo** [comment]
Hey @renanbastos93 this is my first PR in github and I'm not sure what should I do next but I've send a new version in my fork, [can you check it plz?](https://github.com/aqrojo/image-to-base64)

**Comment by renanbastos93** [comment]
I'm sorry I made you wait.
We need to fix a small things.

This lines
 - https://github.com/aqrojo/image-to-base64/blob/3bcc6dcfc064b94d5c3c95fb74bd45000d41b1a1/package.json#L45

File: browser.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+function imageToBase64Browser(urlOrImage, param) {
+    if (!('fetch' in window && 'Promise' in window)) {
+        return Promise.reject('[*] It\'s image2base64 not compatible with your browser.');
+    }
+    return fetch(urlOrImage, param || {}).then(function(response) {
+        return response.arrayBuffer();
+    }).then(base64ToBrowser);
+}
+
+module.exports = imageToBase64Browser;
```
Comment (renanbastos93) [inline browser.js:16]: hey dude, in the browser doesn't exist `module.exports` is a mistake.

lease we need to keep according to example below
```js
(function(escope){
    "use strict";
     // your code ...
     escopoe.imageToBase64 = imageToBase64Browser
})(this)
```

File: package.json [hunk 3]
```diff
@@ -38,6 +42,7 @@
     "node-fetch": "^1.7.3"
   },
   "devDependencies": {
+    "uglify-js": "^3.9.1"
```
Comment (renanbastos93) [inline package.json:45]: please add `,` 
```suggestion
    "uglify-js": "^3.9.1",
```

File: image-to-base64.js [hunk 4]
```diff
@@ -1,72 +1,6 @@
...
-    } else {
-        escope.imageToBase64 = imageToBase64;
-    }
+'use strict';
 
-})(this);
+module.exports = {
+    server: require('./server.min.js'),
+    browser: require('./browser.min.js'),
+};
```
Comment (renanbastos93) [inline image-to-base64.js:6]:
```suggestion
module.exports = require('./server.min.js')
```

Comment (aqrojo) [reply image-to-base64.js:6]: If we can't export both files from the index what do you think about to merge index.js and server.js in the same file?

**Comment by aqrojo** [comment]
Solved! I've merged the server code inside the main file

**Comment by renanbastos93** [comment]
> Solved! I've merged the server code inside the main file

I will do some tests later I answer you

**Comment by aqrojo** [comment]
Great news @renanbastos93, I'm really happy to ear that, thanks so much

**Pull request merged**